app/assets/javascripts/models/page/project_page.js.coffee

```diff
@@ -88,6 +88,31 @@ ns.ProjectPage = do (ko,
     hasType: =>
       @query.serialize().bounding_box || @query.serialize().polygon || @query.serialize().point
 
+    showTemporal: =>
+      if @query.serialize().temporal
+
+        m_names = new Array("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec")
+        label = "" 
+        dates = @query.serialize().temporal.split(",")
+        
+        dateStart = moment(dates[0].split("T")[0].split("-")).subtract(1, 'months') # months start at '0', so we need to subtract one
+        dateEnd = moment(dates[1].split("T")[0].split("-")).subtract(1, 'months')
+
+        if dateStart.year() == dateEnd.year()
+          if dateStart.month() == dateEnd.month()
+            # format only showing the month and the year a single time (e.g., Jan 1 - 10, 2011)
+            label = dateStart.format("MMM DD") + " - " + dateEnd.format("DD, YYYY")
+          else
+            # format only showing the year a single time (e.g., Jan 1 - Feb 10, 2011)
+            label = dateStart.format("MMM DD") + " - " + dateEnd.format("MMM DD, YYYY")
+        else
+          # Otherwise, show the full date range (e.g., Jan 1, 2010 - Feb 10, 2011)
+          label = if dateStart.isValid() then dateStart.format("MMM DD, YYYY") else "Beginning of time "
+          label += if dateEnd.isValid() then " - " + dateEnd.format("MMM DD, YYYY") else " - End of time"
+        label
+      else
+        false
```

Comment:
in the `if` portion of this method the return value is a string, yet in the `else` portion here its a boolean. 

We should ensure that the responses of our methods are consistent types so that when we call them we know what to expect.